compiler/resolution/callDestructors.cpp

```diff
@@ -438,8 +542,44 @@ void ReturnByRef::transformMove(CallExpr* moveExpr)
 
   Expr*     lhs      = moveExpr->get(1);
   CallExpr* callExpr = toCallExpr(moveExpr->get(2));
+  CallExpr* copyExpr = NULL;
   Symbol*   useLhs   = toSymExpr(lhs)->var;
   Symbol*   refVar   = newTemp("ret_to_arg_ref_tmp_", useLhs->type->refType);
+  FnSymbol* fn       = callExpr->isResolved();
+
+  if (fn == NULL)
```

Comment:
On Dec 1, 2015, at 4:49 PM, Kyle Brady <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/resolution/callDestructors.cpp<https://github.com/chapel-lang/chapel/pull/2956#discussion_r46363630>:

>    Symbol*   useLhs   = toSymExpr(lhs)->var;
>    Symbol*   refVar   = newTemp("ret_to_arg_ref_tmp_", useLhs->type->refType);
> +  FnSymbol* fn       = callExpr->isResolved();
> +
> +  if (fn == NULL)


I like the second form you have there, its clearer to me what is going on at least.

—
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/2956/files#r46363630>.


Thanks, I think that’s right.  I hesitated on the same section when I was preparing
the PR.

Mike